flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/ResultPartitionManager.java

```diff
@@ -79,8 +128,47 @@ public ResultSubpartitionView createSubpartitionView(
         return subpartitionView;
     }
 
+    @Override
+    public ResultSubpartitionView createSubpartitionViewOrNotify(
+            ResultPartitionID partitionId,
+            int subpartitionIndex,
+            BufferAvailabilityListener availabilityListener,
+            PartitionRequestNotifier notifier) throws IOException {
+
+        final ResultSubpartitionView subpartitionView;
+        synchronized (registeredPartitions) {
+            final ResultPartition partition = registeredPartitions.get(partitionId);
+
+            if (partition == null) {
+                requestPartitionNotifiers.computeIfAbsent(partitionId, key -> new InputRequestNotifierManager()).addNotifier(notifier);
+                return null;
+            }
+
+            LOG.debug("Requesting subpartition {} of {}.", subpartitionIndex, partition);
+
+            subpartitionView =
+                    partition.createSubpartitionView(subpartitionIndex, availabilityListener);
+        }
+
+        return subpartitionView;
+    }
+
+    @Override
+    public void releasePartitionRequestNotifier(NettyPartitionRequestNotifier notifier) {
```

Comment:
What is the previous behavior when the `ViewReader` was released? Should we send an error to the downstream here?

Reply:
We remove the instance from `ResultPartitionManager` here, and the behavior of reader will be consistent with the original in `CreditBasedSequenceNumberingViewReader.releaseAllResources`